client/src/components/pages/RobotFile/RobotFile.jsx

```diff
@@ -35,16 +55,61 @@ const RobotFile = () => {
    * This function will retrieve the code from the editor, parse it to a ssot and write the
    * resulting ssot into the sessionStorage.
    */
-  const onSaveToCloud = async () => {
-    // eslint-disable-next-line no-unused-vars
-    const robotId = JSON.parse(sessionStorage.getItem('robotId'));
+  const onSaveToCloud = () => {
+    const ssot = parseRobotCodeToSsot(code);
+    if (typeof ssot === 'undefined') {
+      customNotification('Warning', 'Because a parsing error occurred, the robot was not saved to cloud.')
+    } else {
+      sessionStorage.setItem('ssotLocal', JSON.stringify(ssot));
 
-    /**
-     * please parse here to ssot and upsert
-     * use the variables robotId and code for parsing.
-     */
+      upsert();
+    }
   };
 
+  const dataSource = [
```

Comment:
Please use a different file for this if possible. Maybe we could extract the whole modal?